x-pack/plugins/cases/public/components/visualizations/actions/convert_to_absolute_time_range.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TimeRange } from '@kbn/data-plugin/common';
+import { getAbsoluteTimeRange } from '@kbn/data-plugin/common';
+
+export const convertToAbsoluteTimeRange = (timeRange?: TimeRange): TimeRange | undefined => {
+  if (!timeRange) {
+    return;
+  }
+
+  const absRange = getAbsoluteTimeRange(
+    {
+      from: timeRange.from,
+      to: timeRange.to,
+    },
+    { forceNow: new Date() }
+  );
+
+  return {
+    from: absRange.from,
+    to: absRange.to,
+  };
```

Comment:
Should we include the `mode` in the returned object?

Reply:
Good point! I did not notice that. We do not persist the `mode` to the cases SO so I think we should not added here.